src/solc_0.7/catalyst/AssetMinter.sol

```diff
@@ -216,4 +216,15 @@ contract AssetMinter is WithMetaTransaction {
     ) internal {
         _gemsCatalystsRegistry.burnCatalyst(from, catalystId, numTimes * CATALYST_UNIT);
     }
+
+    /// @dev Scale up each number in an array of quantities by a factor of 1000000000000000000.
+    /// @param quantities The array of numbers to scale.
+    /// @return scaledQuantities The scaled-up values.
+    function scaleQuantities(uint256[] memory quantities) internal pure returns (uint256[] memory scaledQuantities) {
```

Comment:
I wonder what would be the gas saving if we did the scale on demand and so where the gemQuantities were already multiplied on the input

Reply:
@wighawag do you mean to do the scaling before calling the contract(ie: in javascript), or just to avoid using an. internal function and perform the calling inside `_batchBurnCatalysts` and `_batchBurnGems` ?

Reply:
yes, the first, basically dealing with token amount